contracts/RRUtils.sol

```diff
@@ -152,4 +152,71 @@ library RRUtils {
 
         return false;
     }
+
+    function compareLabel(bytes memory self, bytes memory other) internal pure returns (int){
+        uint sLength = labelCount(self, 0);
+        uint oLength = labelCount(other, 0);
+        uint shortest;
+
+        if (sLength < oLength){
+            shortest = sLength;
+        }else{
+            shortest = oLength;
+        }
+
+        bytes memory sTail;
```

Comment:
Can you modify these functions to just take offsets, instead of copying the names each time?